src/realm/util/file.cpp

```diff
@@ -256,14 +230,27 @@ void remove_dir(const std::string& path)
 
 bool try_remove_dir(const std::string& path)
 {
-#ifdef _WIN32
-    std::wstring w_path = string_to_wstring(path);
-    if (_wrmdir(w_path.c_str()) == 0)
-        return true;
+#if REALM_HAVE_STD_FILESYSTEM
+    std::error_code error;
+    bool result = std::filesystem::remove(path, error);
+
+    if (!error) {
+        return result;
+    }
+
+    using std::errc;
+    if (error == errc::permission_denied || error == errc::read_only_file_system ||
+        error == errc::device_or_resource_busy || error == errc::operation_not_permitted ||
+        error == errc::file_exists || error == errc::directory_not_empty) {
+        throw File::PermissionDenied(error.message(), path);
```

Comment:
It feels weird that we have different definitions of "permission denied" here than on line 158. It seems like we should be able to have a `throwFileExceptionIfFailed(std::error_code)` function to handle this. If you do that, it could replace the `if (!error)` on line 237 as well, so you could just do `throwIfFailed(error); return result;`

Edit: I see that there are a ton of blocks like this. Unless they *really* need different checks, please pull this out to a function.

Reply:
Very good point. I've refactored these blocks out to a helper function.